app/controllers/image_shares_controller.rb

```diff
@@ -3,15 +3,11 @@ class ImageSharesController < ApplicationController
 
   before_action :find_image
 
-  def new; end
-
   def create
     ImageMailer.share_image_email(@image, share_params[:email], share_params[:message]).deliver_now
     flash[:success] = 'You have successfully shared an image.'
```

Comment:
Since this action is triggered by an XHR request, I think we'd better not use `html` way to handle the response. Instead, we could use `respond_to`. Do you guys want to update it?

Reply:
@HaominZhang, we've updated the handling of the XHR request to use JavaScript to render the flash message, rather than relying on the `ajax:complete` message to trigger a page reload.